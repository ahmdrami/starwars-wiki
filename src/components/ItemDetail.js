import React from 'react';


const ItemDetail = ({ item, path }) => {
   return (
      <main>
         {!item ? (
            <h1 className="title"> { path === 'paths' ? 'Star Wars Resources' : path } </h1>
         ) : (
            <section>
               <span className="subtitle"> { path }</span>
               <h1 className="title">{ item.name || item.title }</h1>
               {Object.keys(item)
               .filter(filterKey => filterKey !== 'name' )
               .map((itemKey, i) => (
                  <div className="item-container" key={i}>
                     <label> {itemKey} </label>
                     <span> {item[itemKey]} </span>
                  </div>
               ))}
            </section>
         )}
      </main>
   );
};

export default ItemDetail;
