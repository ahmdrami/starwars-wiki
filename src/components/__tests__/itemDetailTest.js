import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import ItemDetail from '../ItemDetail';

configure({ adapter: new Adapter() });
const testProps = {
   item: null,
   path: 'paths'
};
describe('ItemDetail Component', () => {
   it('Should display Start Wars Resources when item is null', () => {
      const wrapper = shallow(<ItemDetail {...testProps} />);
      expect(wrapper.text()).toContain('Star Wars Resources');
      expect(wrapper).toMatchSnapshot();

   });

   it('Should create 2 item containers and do not include the name', () => {
      testProps.item = {
         name: 'Test',
         age: '44',
         height: '6'
      }
      const wrapper = shallow(<ItemDetail {...testProps} />);
      expect(wrapper.find('.item-container').length).toBe(2);

   });

});
