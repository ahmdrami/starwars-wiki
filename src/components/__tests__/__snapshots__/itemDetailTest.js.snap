// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail Component Should display Start Wars Resources when item is null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemDetail
    item={null}
    path="paths"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="title"
>
         
        Star Wars Resources
         
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "Star Wars Resources",
          " ",
        ],
        "className": "title",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "Star Wars Resources",
        " ",
      ],
      "type": "h1",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="title"
>
           
          Star Wars Resources
           
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Star Wars Resources",
            " ",
          ],
          "className": "title",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Star Wars Resources",
          " ",
        ],
        "type": "h1",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
